catalog/source/iwc_manifest_to_workflows_yaml.py

```diff
@@ -0,0 +1,180 @@
+        for workflow in repo["workflows"]:
+            if not "tests" in workflow:
+                # probably fixed on main branch of iwc ?
+                # this branch is pretty out of date
+                continue
+            workflow_input = WorkflowInput(
+                active=False,
+                trs_id=f'{workflow["trsID"]}/versions/v{workflow["definition"]["release"]}',
+                workflow_name=workflow["definition"]["name"],
+                categories=workflow["collections"],
+                workflow_description=workflow["definition"]["annotation"],
+                ploidy="ANY",
+                # readme=workflow["readme"],
+                # shortcut so we don't need to parse out the whole inputs section
+                parameters=get_input_types(workflow["definition"]),
+            )
+            by_trs_id[workflow["trsID"]] = workflow_input
+    return by_trs_id
+
+
+def merge_into_existing():
+    existing = read_existing_yaml()
+    current = generate_current_workflows()
+    merged: Dict[str, WorkflowInput] = {}
+    for versionless_trs_id, current_workflow_input in current.items():
+        existing_workflow_input = existing.get(versionless_trs_id)
+        if existing_workflow_input:
+            # someone has annotated ploidy and parameters, keep these, but also check
+            # that existing parameters still exist
+            current_workflow_input.ploidy = existing_workflow_input.ploidy
+            current_workflow_input.active = existing_workflow_input.active
+            # check that specified parameters still exist
+            for param_key in existing_workflow_input.parameters:
+                if param_key not in current_workflow_input.parameters:
+                    # Should be rare, but can happen.
+                    raise Exception(
+                        f"{param_key} specified but is not part of updated workflow {current_workflow_input.trs_id}! Review and fix manually"
+                    )
+            # keep existing parameters
+            current_workflow_input.parameters = existing_workflow_input.parameters
+        merged[versionless_trs_id] = current_workflow_input
+    return merged
+
+
+def to_workflows_yaml(exclude_other: bool):
+    by_trs_id = merge_into_existing()
+    # sort by trs id, should play nicer with git diffs
+    sorted_workflows = [w.as_dict() for w in dict(sorted(by_trs_id.items())).values()]
+    if exclude_other:
+        print("excluding!")
+        final_workflows = []
+        for workflow in sorted_workflows:
+            if "OTHER" in workflow["categories"]:
+                workflow["categories"].remove("OTHER")
+                if not workflow["categories"]:
+                    print(f'Excluding workflow {workflow["trs_id"]}, category unknown')
+                    continue
+            final_workflows.append(workflow)
+        sorted_workflows = final_workflows
+    final_workflows = sorted_workflows
+    with open(WORKFLOWS_PATH, "w") as out:
```

Comment:
so if im understanding so far, the new flow is to overwrite the yml file w this script, then run the existing script to make workflows.json from the now semi-manually, semi-programmatically created yaml? i wonder if a better flow would be something like to put a list of only the workflows we want in workflows yml w only the fields that need manual curation (ploidy, soon to be tax id), and then add the params, descriptions, names, etc as part of the script that produces the json?

Reply:
also, is this automatically picking up new versions of workflows? is that a thing we want?

Reply:
I don't see what the advantage is with your suggestion. As a maintainer you look at a single file, set what's needed to be set and you toggle the active flag. Updates will work fine for the most part unless a specified parameter has changed, in which case the script fails and you have to fix this manually.

Yes, we want the latest versions, this is a safe way to do what https://github.com/galaxyproject/brc-analytics/issues/277 was aiming at.